src/lm_buddy/jobs/_entrypoints/prometheus.py

```diff
@@ -0,0 +1,165 @@
+    """
+
+    return client.completions.create(
+        model=config.prometheus.inference.engine,
+        prompt=prompt,
+        best_of=config.prometheus.best_of,
+        max_tokens=config.prometheus.max_tokens,
+        frequency_penalty=config.prometheus.frequency_penalty,
+        temperature=config.prometheus.temperature,
+        top_p=config.prometheus.top_p,
+    )
+
+
+def parse_response(config: PrometheusJobConfig, response: Completion) -> tuple[str, str]:
+    """Given a Prometheus eval response as returned by the OpenAI API
+    endpoint (i.e. in Completion format), extract feedback
+    and score.
+    """
+
+    if response is None:
+        raise BadResponseError("Server returned an empty response")
+
+    try:
+        response_text = response.choices[0].text
+        # note: this can raise a ValueError if the message is malformed
+        feedback, score = response_text.split("[RESULT]")
+        feedback = feedback.strip()
+        score = score.strip()
+        if score not in [
+            str(s) for s in range(config.evaluation.min_score, config.evaluation.max_score + 1)
+        ]:
+            raise BadResponseError(f"Score {score} is not in range")
+    except (ValueError, BadResponseError) as e:
+        raise BadResponseError(f"Server returned a malformed response ({e})", e)
+
+    return feedback, score
+
+
+def instruction_to_prompt(instruction: str) -> str:
+    """Given some text containing Prometheus instructions, transform it
+    into an actual llama-2 prompt.
+    """
+    conv = get_conv_template("llama-2")
+    conv.set_system_message("You are a fair evaluator language model.")
+    conv.append_message(conv.roles[0], instruction)
+    conv.append_message(conv.roles[1], None)
+    return conv.get_prompt()
+
+
+def run_eval(
+    config: PrometheusJobConfig, data: Dataset, tokenizer: PreTrainedTokenizer, client: OpenAI
+) -> str:
+    # enable / disable tqdm
+    dataset_iterable = tqdm(data) if config.evaluation.enable_tqdm else data
+
+    # open the output file for writing and iterate on samples
+    tracking_name = config.tracking.name if config.tracking is not None else "output.json"
+    output_fname = Path(config.evaluation.tmp_folder) / tracking_name
+    with output_fname.open("w") as file:
+        for sample in dataset_iterable:
+            # convert instructions from the dataset (`text_field` in a dict) to
+            # prompts that prometheus accepts
+            prompt = instruction_to_prompt(sample[config.dataset.text_field])
+
+            # skip those examples which are too long
+            tokenized_prompt = tokenizer(prompt, truncation=False)
+            if len(tokenized_prompt["input_ids"]) > 3072:
+                continue
+
+            # prepare output
+            result = copy.deepcopy(sample)
+            result["prometheus_output"] = []
+            result["prometheus_score"] = []
+
+            for idx in range(config.evaluation.num_answers):
+                i = 0
```

Comment:
could we rename `i` to something more descriptive, like `current_retry_attempt` for troubleshooting?

Reply:
Retries include both openai calls and result parsing, as this could be malformed for many different reasons.
I will factor both of them out!